master_server/dummy_vm_server.py

```diff
@@ -0,0 +1,104 @@
+"""
+  This is a dummy VM server.
+  It is built to test master server.
+"""
+import sys
+import os
+import threading
+from datetime import datetime
+import requests
+from flask import Flask
+from flask import request
+
+
+class MyFlaskApp(Flask):
+  """
+    This class configures the flask app and executes function
+    pre_task before starting flask server.
+  """
+  def run(self, host=None, port=None, debug=None, load_dotenv=True, **options):
+    if not self.debug or os.getenv('WERKZEUG_RUN_MAIN') == 'true':
+      with self.app_context():
+        pre_task()
+    super(MyFlaskApp, self).run(
+        host=host, port=int(sys.argv[1]), debug=debug,
+        load_dotenv=load_dotenv, **options)
+
+app = MyFlaskApp(__name__)
+
+flag = False
+master_server = 'http://127.0.0.1:5000'
+vm_address = 'http://127.0.0.1:'
+request_id = ''
+
+def task_done():
+  """
+    It prints time and then sleep for some time
+    and then again print time.
+  """
+  global flag
+  now = datetime.now()
+  current_time = now.strftime('%H:%M:%S')
+  print('Current Time =', current_time)
+  os.system('sleep 10')
```

Comment:
Let's make the sleep random so that it sometimes exceeds the deadline and sometimes doesn't?

Reply:
Changed sleep time to random value